### PR TITLE
Add JinjaReader to read external template files

### PR DESCRIPTION
**Pull request by ljvmiranda921**
## Description

This Pull Request adds a `spacy.JinjaReader.v1` registered in the `miscs` registry so that we can pass external `.jinja2` templates in the files. The function signature returns a string instead of a `jinja2.Template`, so the "actual" creation of a template that can be rendered happens downstream. 

### Types of change

- New custom function

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by ljvmiranda921** [comment]
@rmitsch , I'm hitting a `FileNotFoundError` again just like the last time. Did we do something special to fix the problem with the `EXAMPLES_DIR`? I can't seem to surmise it in the codebase. 🤔

**Comment by adrianeboyd** [comment]
There's a list of file endings in `MANIFEST.in` for files that should be included in the package.

File: README.md
Comment (ljvmiranda921) [inline README.md:261]: Unsure of this. There is a default, actually, the `_DEFAULT_NER_TEMPLATE`. Not sure what to put in the `Default` column.

File: spacy_llm/tasks/ner.py
Comment (ljvmiranda921) [inline spacy_llm/tasks/ner.py:115]: Decided to just raise an error when `template` is `None` rather than assigning default. Not too attached with this decision though. Happy to hear alternatives!

**Comment by ljvmiranda921** [comment]
Ready for review! Note that this is dependent on #81 (it merges to that branch) cc: @rmitsch @kabirkhan

File: README.md
Comment (rmitsch) [reply README.md:261]: Hm, we could move _DEFAULT_NER_TEMPLATE into a registry function and specify the registry handle of that function here? Although it seems odd to do this just for the docs. Alternatively a "See here" linking to `_DEFAULT_NER_TEMPLATE` in the code?

File: README.md
Comment (rmitsch) [inline README.md:319]: Same w.r.t. default value as above.

**Review by rmitsch — COMMENTED** [review]
Looks good! There'll be more things to test in terms of path resolution and (de-)serialization, but I think this is fine for now. Just minor comments.

I'll merge this after https://github.com/explosion/spacy-llm/pull/81 to keep the diff there clean.

File: README.md
Comment (ljvmiranda921) [reply README.md:261]: I like the latter suggestion!

**Comment by ljvmiranda921** [comment]
Thanks for the reviews! I'll fix the conflicts once #81 's been merged! 🙇 (so it's just a one time conflict fix)

File: spacy_llm/registry/reader.py
Comment (adrianeboyd) [inline spacy_llm/registry/reader.py:9]: I feel like this has moved away from how it was originally envisioned and should  probably be turned into a more generic file reader instead? All it does is require a particular file ending and then read in a string.

And in terms of the file extensions it seems overly restrictive for jinja where there are multiple file endings in common use? (spacy uses `.jinja`)

File: spacy_llm/ty.py
Comment (ljvmiranda921) [reply spacy_llm/ty.py:15]: Oh I admit I was just copying the `ExamplesConfigType` here. I thought this means: accept any string and a `JinjaReader` (which is a `Callable` that returns a string). Sometimes, I'm still confused by types so let me know what's the appropriate thing to do!

File: spacy_llm/tasks/ner.py
Comment (rmitsch) [inline spacy_llm/tasks/ner.py:58]: IMO `template` should be resolved by the config system (if e. g. `spacy.FileReader.v1` is set), so I don't think we ever get a `Callable` here?

File: spacy_llm/tests/tasks/test_textcat.py
Comment (rmitsch) [inline spacy_llm/tests/tasks/test_textcat.py:181]:
```suggestion
    """Use OpenAI to get Textcat results.
```

File: spacy_llm/tests/tasks/test_textcat.py
Comment (rmitsch) [inline spacy_llm/tests/tasks/test_textcat.py:183]:
```suggestion
    to be consistent/predictable.
```

File: spacy_llm/registry/reader.py
Comment (rmitsch) [inline spacy_llm/registry/reader.py:11]:
```suggestion
    """Read a file from a path and return its contents.
```

File: spacy_llm/tasks/rel.py
Comment (rmitsch) [inline spacy_llm/tasks/rel.py:73]: Same here.

File: spacy_llm/tasks/spancat.py
Comment (rmitsch) [inline spacy_llm/tasks/spancat.py:57]: Same here.

**Review by rmitsch — COMMENTED** [review]
Almost good to go.

File: spacy_llm/tasks/ner.py
Comment (kabirkhan) [reply spacy_llm/tasks/ner.py:58]: fixing this

Comment (kabirkhan) [reply spacy_llm/tasks/ner.py]: Ok I removed this and also just remove the error if template is None. We have type hints here for a reason + we have a default template for each task so the user would need to explicitly set the template to None

**Comment by ljvmiranda921** [comment]
From what I can tell from the diffs the big change here was removing the `Callable` and just using `str`. I think it looks good! Ok for merging later :slightly_smiling_face:

**Review by rmitsch — APPROVED** [review]

**Pull request merged**
